protoc-gen-openapiv2/internal/genopenapi/template.go

```diff
@@ -782,15 +782,46 @@ func isResourceName(prefix string) bool {
 	return field == "parent" || field == "name"
 }
 
-func renderServices(services []*descriptor.Service, paths openapiPathsObject, reg *descriptor.Registry, requestResponseRefs, customRefs refMap, msgs []*descriptor.Message) error {
+func renderServiceTag(svc *descriptor.Service) []openapiTagObject {
```

Comment:
Is there a good reason this returns a slice rather than just a `openapiTagObject`?
```suggestion
func renderServiceTag(svc *descriptor.Service) openapiTagObject {
```

Reply:
Obviated by factoring service tag rendering into it's own loop as suggested in other comments.